src/main/java/com/commilitio/medicalclinic/repository/PatientRepository.java

```diff
@@ -0,0 +1,91 @@
+package com.commilitio.medicalclinic.repository;
+
+import com.commilitio.medicalclinic.model.Patient;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+
+@Getter
+@RequiredArgsConstructor
+@Repository
+public class PatientRepository {
+
+    private final List<Patient> patientList;
+
+    public Optional<Patient> getPatient(String email) {
+        return Optional.of((Patient) patientList.stream()
+                .filter(patient -> patient.getEmail().equals(email)));
+    }
+
+    public Optional<Patient> addPatient(Patient patient) {
+        boolean patientExists = patientList.stream()
+                .anyMatch(existingPatient -> existingPatient.getIdCardNo().equals(patient.getIdCardNo()));
+
+        if (patientExists) {
+            System.out.println("Patient already exists.");
+            return Optional.empty();
+        } else {
+            patientList.add(patient);
+            System.out.println("Patient successfully added.");
+            return Optional.of(patient);
+        }
+    }
+
+    public void deletePatient(String email) {
```

Comment:
tutaj w sytuacji kiedy pacjenta o danym mailu nie bedzie istniaj to nie zostanie on usuniety a uzytkownik dostanie http status 200 sugerujac ze operacja się wykonała prawidłowo